src/App.js

```diff
@@ -10,10 +11,33 @@ const tabs = [
 ];
 /* eslint-enable */
 
-const App = () => (
-  <div className="App">
-    <h1>Selected tab is&nbsp;Tab 1</h1>
-  </div>
-);
+class App extends React.Component {
+  state = {
+    activeTab: tabs[0],
+  }
+
+  handlerEvent = (id) => {
+    this.setState(() => ({
+      activeTab: tabs.find(tab => tab.id === id),
+    }));
```

Comment:
You don't use previous value, so no need to use functional `setState` here.
Also, if id equals active tab id - setState should not be called.